src/main/java/pro/trevor/tankgame/rule/impl/version4/AttributeDecoder.java

```diff
@@ -0,0 +1,64 @@
+package pro.trevor.tankgame.rule.impl.version4;
+
+import org.json.JSONObject;
+import pro.trevor.tankgame.state.board.unit.tank.IAttributeDecoder;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class AttributeDecoder implements IAttributeDecoder<TankAttribute> {
```

Comment:
This duplicates a lot of code from the version 3 AttributeDecoder could it be made generic?